sdk/src/main/scala/com/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -403,36 +376,18 @@ class EthService(
 
   @RpcMethod("eth_sendRawTransaction") def sendRawTransaction(signedTxData: String): Quantity = {
     val tx = new EthereumTransaction(EthereumTransactionDecoder.decode(signedTxData))
-    validateAndSendTransaction(tx)
+    implicit val timeout: Timeout = Timeout.apply(5, SECONDS)
+    Await.result(sidechainTransactionActorRef ? BroadcastTransaction(tx), timeout.duration)
     new Quantity(Numeric.prependHexPrefix(tx.id))
   }
 
-  private def validateAndSendTransaction(
-      transaction: Transaction,
-      transactionResponseRepresentation: Transaction => SuccessResponse = defaultTransactionResponseRepresentation
-  ) = {
-    implicit val timeout: Timeout = 5 seconds
-    val barrier = Await
-      .result(sidechainTransactionActorRef ? BroadcastTransaction(transaction), timeout.duration)
-      .asInstanceOf[Future[Unit]]
-    onComplete(barrier) {
-      // TODO: add correct responses
-      case Success(_) => ApiResponseUtil.toResponse(transactionResponseRepresentation(transaction))
-      case Failure(exp) =>
-        ApiResponseUtil.toResponse(GenericTransactionError("GenericTransactionError", JOptional.of(exp)))
```

Comment:
It seems that we don't cover this case anymore

Reply:
- If I'm not mistaken `ApiResponseUtil.toResponse()` is the wrong tool to use within the RPC api as it would write to the response directly, bypassing our RPC response serialization.
- Also I expected `Await.result` to throw in case the future does not resolve - and exceptions are caught further up and returned as an RPC error response. Did not test this though.

Reply:
Fixed and added a simple python test that checks that an invalid TX is rejected with an error, instead of getting a success response with the tx hash.